Projects/Server/Common/Filters/Filters.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+
+namespace Server.Common.Filters;
+
+public readonly struct DummyFilter<T> : IFilter<T, T>
+{
+    public T? Invoke(T instance)
+    {
+        return instance;
+    }
+}
+
+public readonly struct ChainFilter<TIn, TMiddle, TOut, TFirst, TSecond> : IFilter<TIn, TOut>
```

Comment:
I love the idea of struct filters. Let's do two things:

1. Move the filters to their own set of files since they should be generic enough to be used outside of just containers.
2. Let's benchmark this. We have a benchmark repo (which probably needs updating, I can do that later). I would love to test out multiple approaches.